vm/src/sliceable.rs

```diff
@@ -315,6 +315,19 @@ pub fn saturate_index(p: isize, len: usize) -> usize {
     p as usize
 }
 
+// Saturate p in range [0, len] inclusive
+pub fn pyint_saturate_index(p: PyIntRef, len: usize) -> usize {
```

Comment:
The general approach is perfect, but I need time to think where will be the best place to put this function